src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,278 @@
+  // but this should be good enough for most use cases.
+  #keyupListener = (e) => {
+    const { key } = e;
+    // only cancel on Escape
+    if (key === 'Escape') {
+      this.removeEventListener('keyup', this.#keyupListener);
+      return;
+    }
+
+    if (key === 'Enter' || key === ' ') {
+      this.handleSelection(e);
+    } else {
+      this.handleMovement(e);
+    }
+  }
+
+  #keydownListener = (e) => {
+    const { metaKey, altKey, key } = e;
+    if (metaKey || altKey) {
+      this.removeEventListener('keyup', this.#keyupListener);
+      return;
+    }
+    this.addEventListener('keyup', this.#keyupListener, {once: true});
+  }
+
+  enable() {
+    this.addEventListener('click', this.#clickListener);
+    this.addEventListener('keydown', this.#keydownListener);
+  }
+
+  disable() {
+    this.removeEventListener('click', this.#clickListener);
+    this.removeEventListener('keyup', this.#keyupListener);
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_CONTROLLER) {
+      if (oldValue) {
+        const mediaControllerEl = document.getElementById(oldValue);
+        mediaControllerEl?.unassociateElement?.(this);
+      }
+      if (newValue) {
+        const mediaControllerEl = document.getElementById(newValue);
+        mediaControllerEl?.associateElement?.(this);
+      }
+    } else if (attrName === 'disabled' && newValue !== oldValue) {
+      if (newValue == null) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+    }
+  }
+
+  connectedCallback() {
+    if (!this.hasAttribute('disabled')) {
+      this.enable();
+    }
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.associateElement?.(this);
+    }
+  }
+
+  disconnectedCallback() {
+    this.disable();
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get keysUsed() {
+    return ['Enter', ' ', 'ArrowDown', 'ArrowUp', 'Home', 'End'];
+  }
+
+  #getItem(e) {
+    if (e.target === this) return;
+
+    const composedPath = e.composedPath();
+    const index = composedPath.findIndex(el => el.nodeName === 'MEDIA-CHROME-LISTITEM');
+
+    return composedPath[index];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
+    if (!item) return;
+
+    const selected = item.getAttribute('aria-selected') === 'true';
+
+    if (this.getAttribute('aria-multiselectable') !== 'true') {
+      this.#assignedElements.forEach(el => el.setAttribute('aria-selected', 'false'));
+    }
+
+    if (selected) {
+      item.setAttribute('aria-selected', 'false');
```

Comment:
This seems to assume that you can turn off a selection and have no item selected. Thinking of something like the  playback rate menu, you can't turn off playback rate. Something always has to be selected. Should we be making that an option at this level, or should whatever's using this handle that?

Reply:
I think that would happen at a higher level, like the playback rate menu, rather than here. Because it's reasonable to be able to unselect things for a generic listbox.